### ISSUE TITLE
Android flickering

### ISSUE THREAD

**Issue by odusseys** [react-community/react-native-webgl#33]
I'm linking to this issue I posted on the Expo repo: 

https://github.com/expo/expo/issues/868

The issue is flickering on Android when the js is too slow for the onDrawFrame method of the GLSurfaceView. 
We need some kind of clean synchronization between the JS and Java (see the suggestion I posted there).

**Comment by odusseys** [comment]
I have a fix for this but it is a bit dirty : 

- set the renderMode to "RENDERMORE_WHEN_DIRTY"
- do a requestRender() when calling endFrame (have to add a method to the bridge)
- because some of the gl calls are blocking, we need to flush them regularly, and the above will cause a deadlock. The way I solved that is to add another thread which, in a loop, calls queueEvent on the GLSurfaceView, requesting a flush (btw, I don't see why we don't also do this for textures).

This makes it work, with a few caveats: 
- it seems to break textures again :( might be another concurrency issue, looking into it
- Even with WHEN_DIRTY, the GLSurfaceView may perform a few onDrawFrame() calls before it "settles". This means that 
  1) It may cause sync issues when initializing the view (gl events being flushed to the wrong frame)
  2) When the app is brought to the background, a new frame is drawn and thus blank. 
Needs thoughts.

**Comment by odusseys** [comment]
can confirm that using queueEvent instead of using the custom queue (mEventQueue) fixes texture problems, and simplifies the code greatly. Only the second point is a problem.

**Comment by odusseys** [comment]
you can look at the work on this fork : https://github.com/fitle-dev/react-native-webgl

**Comment by DylanVann** [comment]
Might fix this: https://github.com/react-community/react-native-webgl/pull/62

**Comment by Longhin-Mattia** [comment]
[#62](https://github.com/react-community/react-native-webgl/pull/62) does not work for me.

- I cloned the repo
- Replaced RNWebGLView.java with the new one
- Updated react-native package to be compatible with JavaScriptContextHolder

Flickering is still there! :-(

**Comment by DylanVann** [comment]
I'll mention that I specifically had an issue with flickering after putting the phone to sleep and re-waking it. The flickering was flickering between a black screen and what I was trying to draw.

I'm now using my own fork with this fix and that no longer happens.

Maybe a more detailed description of the issue would help.

**Comment by Longhin-Mattia** [comment]
@DylanVann: Thank you for your feedback.

The issue is simple: there is a very annoying flickering between a black screen and the image. I can see this in emulator and real device.

- I tried with a fresh new react-native app (just `react-native init Test`  and `npm install react-native-webgl [...]`)

- I also tried to compile the example app of the repository. It does not work either.

I applied the modification of your PR but it does not solve the issue. I'm currently trying to use the old [gl-react-native-v2](https://github.com/gre/gl-react-native-v2) to see if it works better.

Did anyone faced the same problem and found a solution?
